cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java

```diff
@@ -83,6 +87,40 @@ public StatefulSetOperator(Vertx vertx, KubernetesClient client, long operationT
         return f;
     }
 
+    public Future<Void> maybeCleanPod(StatefulSet ss) {
+        String namespace = ss.getMetadata().getNamespace();
+        String name = ss.getMetadata().getName();
+        final int replicas = ss.getSpec().getReplicas();
+        log.debug("Considering manual deletion and restart of pods for {}/{}", namespace, name);
+        Future<Void> f = Future.succeededFuture();
+        for (int i = 0; i < replicas; i++) {
+            String podName = name + "-" + i;
+            String pvcName = AbstractModel.getPersistentVolumeClaimName(name, i);
+            Pod pod = podOperations.get(namespace, podName);
+            String value = pod.getMetadata().getAnnotations().get(ANNOTATION_MANUAL_POD_CLEANING);
+            if (value != null && Boolean.valueOf(value)) {
+                f = f.compose(ignored -> maybeDeletePvc(ss, pvcName))
```

Comment:
Do we need some form of validation that we are using persistent storage? Or will this work fine without errors on ephemeral?

Reply:
Yeah, it's working without any errors. It's just that the resource doesn't exist in that case but no exceptions are thrown.

Reply:
Ok, no problem with that then. :+1: